Assets/Scripts/Cannon.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class Cannon : MonoBehaviour
+{
+    [SerializeField] private float _fovAngle = 80f;
+    [SerializeField] private float _shootCooldown = 1.0f;
+    [SerializeField] private Vector3 _boxCastSize = new Vector3(15f, 15f, 0);
+    [SerializeField] private LayerMask _targetLayers;
+    [SerializeField] private Transform _projectileSpawnPoint;
+    [SerializeField] private float _rotationSpeed = 1.5f;
+    [SerializeField] private Transform _cannonBarrel;
+    [SerializeField] private GameObject _projectileGameObject;
+    [SerializeField] private float _projectileSpeed = 10f;
+
+    private void Start()
+    {
+        _transform = transform;
+        _originalRotation = _transform.rotation;
+    }
+
+    private void Update()
+    {
+        if(_target == null)
+        {
+            var step = _rotationSpeed * Time.deltaTime;
+            transform.rotation = Quaternion.RotateTowards(transform.rotation, _originalRotation, step);
+            
+            var hits = Physics.BoxCastAll(transform.position, _boxCastSize, transform.forward, Quaternion.identity,
+                float.MaxValue, _targetLayers);
+
+            foreach (var hit in hits)
+            {
+                if (!(GetAngle(hit.transform) * 2 < _fovAngle)) continue;
+                _target = hit.collider.gameObject;
+                break;
+            }
+        }
+        else
+        {
+            if ( _targetLayers != (_targetLayers | (1 << _target.layer)))
+            {
+                _target = null;
+                return;
+            }
+            
+            var baseDesiredDirection = _target.transform.position - _transform.position;
+            var baseCross = Vector3.Cross(baseDesiredDirection, _transform.forward);
+            var baseDir = Mathf.Abs(baseCross.y) < 0.05 ? 0 : baseCross.y > 0 ? -1 : 1;
```

Comment:
these 0.05 and 0.1 appear several times throughout this function
consider making them constants or serializable fields